Gems/Atom/RHI/DX12/Code/Source/RHI/CommandList.cpp

```diff
@@ -725,9 +709,30 @@ namespace AZ
             m_state.m_shadingRateState.Set(rate, combinators);
         }
 
-        void CommandList::BuildTopLevelAccelerationStructure([[maybe_unused]] const RHI::RayTracingTlas& rayTracingTlas)
+        void CommandList::BuildTopLevelAccelerationStructure(
+            const RHI::RayTracingTlas& rayTracingTlas, const AZStd::vector<const RHI::RayTracingBlas*>& changedBlasList)
         {
 #ifdef AZ_DX12_DXR_SUPPORT
+            ID3D12GraphicsCommandList4* commandList = static_cast<ID3D12GraphicsCommandList4*>(GetCommandList());
+            if (!changedBlasList.empty())
+            {
+                // create a barrier for BLAS completion
+                // this is required since all BLAS must be built prior to using it in the TLAS
+                std::vector<D3D12_RESOURCE_BARRIER> barriers;
+                barriers.reserve(changedBlasList.size());
+                for (const auto* blas : changedBlasList)
+                {
+                const auto dx12RayTracingBlas = static_cast<const RayTracingBlas*>(blas);
+                const RayTracingBlas::BlasBuffers& blasBuffers = dx12RayTracingBlas->GetBuffers();
+
+                D3D12_RESOURCE_BARRIER barrier;
+                barrier.Type = D3D12_RESOURCE_BARRIER_TYPE_UAV;
+                barrier.Flags = D3D12_RESOURCE_BARRIER_FLAG_NONE;
+                barrier.UAV.pResource = static_cast<Buffer*>(blasBuffers.m_blasBuffer.get())->GetMemoryView().GetMemory();
+                barriers.push_back(barrier);
```

Comment:
Minor: tabbing

Reply:
Fixed this. clang-format does not handle this correctly for some reason